cmake/Modules/VolkPython.cmake

```diff
@@ -112,17 +112,26 @@ execute_process(
     COMMAND ${PYTHON_EXECUTABLE} -c "import os
 import sysconfig
 import site
+
 install_dir = None
 prefix = '${CMAKE_INSTALL_PREFIX}'
-#use sites when the prefix is already recognized
+
+#use `site` when the prefix is already recognized
 try:
   paths = [p for p in site.getsitepackages() if p.startswith(prefix)]
   if len(paths) == 1: install_dir = paths[0]
 except AttributeError: pass
+
 if not install_dir:
-    #find where to install the python module
-    install_dir = sysconfig.get_path('platlib')
-    prefix = sysconfig.get_config_var('prefix')
+    # find where to install the python module
+    # for Python 3.11+, we could use the 'venv' scheme for all platforms
+    if os.name == 'nt':
+        scheme = 'nt'
+    else:
+        scheme = 'posix_prefix'
+    install_dir = sysconfig.get_path('platlib', scheme)
+    prefix = sysconfig.get_path('data', scheme)
```

Comment:
Thinking this should be `prefix`, not `data`, per the prior code ... yes?

Reply:
On Ubuntu 22.04 the options for the first parameter are:
```
>>> sysconfig.get_path_names()
('stdlib', 'platstdlib', 'purelib', 'platlib', 'include', 'scripts', 'data')
```
The results are:
```
path=stdlib              /usr/lib/python3.10
path=platstdlib          /usr/lib/python3.10
path=purelib             /usr/lib/python3.10/site-packages
path=platlib             /usr/lib/python3.10/site-packages
path=include             /usr/include/python3.10
path=scripts             /usr/bin
path=data                /usr
```
with scheme `posix_prefix`. Further, with `data` and `platlib` the resulting relative paths are:
```
scheme=deb_system          lib/python3/dist-packages
scheme=nt                  Lib/site-packages
scheme=nt_user             Python/site-packages
scheme=osx_framework_user  lib/python/site-packages
scheme=posix_home          lib/python
scheme=posix_local         lib/python3.10/dist-packages
scheme=posix_prefix        lib/python3.10/site-packages
scheme=posix_user          lib/python3.10/site-packages
```
Thus, `scheme='posix_prefix'` seems to be fitting. However, it seems like we might want to opt for `scheme='posix_local'` because on Ubuntu 22.04:
```
>>> import sys
>>> sys.path
[...,
'/usr/lib/python310.zip', 
'/usr/lib/python3.10', 
'/usr/lib/python3.10/lib-dynload', 
'/usr/local/lib/python3.10/dist-packages', 
'/usr/lib/python3/dist-packages']
```
On Ubuntu 20.04:
```
[...
'/usr/lib/python38.zip', 
'/usr/lib/python3.8', 
'/usr/lib/python3.8/lib-dynload', 
'/usr/local/lib/python3.8/dist-packages', 
'/usr/lib/python3/dist-packages']
```
Then again, in a `fedora:36` Docker container:
```
[...
'/usr/lib64/python310.zip', 
'/usr/lib64/python3.10', 
'/usr/lib64/python3.10/lib-dynload', 
'/usr/lib64/python3.10/site-packages', 
'/usr/lib/python3.10/site-packages']
```
Basically, we want that Python files get installed into the correct folder for the default `CMAKE_INSTALL_PREFIX` which is `/usr/local`. The installation should work without modifying the `PYTHONPATH` variable.
The variables are: 
- `lib` vs `lib64`
- `dist-packages` vs `site-packages`
- `python` vs `python3` vs `python3.x`.
It seems like our variable choice works for Fedora 36+ at least:
https://docs.fedoraproject.org/en-US/fedora/latest/release-notes/developers/Development_Python/

Reply:
I suggest to use `posix_prefix` and be consistent with GR. Thus, a non-standard install path uses `site-packages`. That would also be the case with Python 3.11+ and `venv`.

Reply:
@jdemel OK ... so the code in the PR as is, no further changes? yes?